File: src/js/AboutV12.js
export function aboutV12() {
  const banner = `
____    ____  __   ___       _______ .__   __.   _______  __  .__   __.  _______
\\   \\  /   / /_ | |__ \\     |   ____||  \\ |  |  /  _____||  | |  \\ |  | |   ____|
 \\   \\/   /   | |    ) |    |  |__   |   \\|  | |  |  __  |  | |   \\|  | |  |__
  \\      /    | |   / /     |   __|  |  . \`  | |  | |_ | |  | |  . \`  | |   __|
   \\    /     | |  / /_     |  |____ |  |\\   | |  |__| | |  | |  |\\   | |  |____
    \\__/      |_| |____|    |_______||__| \\__|  \\______| |__| |__| \\__| |_______|
`;
  print_c(colors.magenta, styles.bold, banner);
  n_line();
  print_c(colors.magenta, styles.bold, "About V12 - A simple, lightweight, and easy-to-use JavaScript Interpreter/Engine written in Rust.");
  print_c(colors.magenta, styles.bold, "Version: 1.0.0");
  print_c(colors.magenta, styles.bold, "Author(s): Thoq");
  print_c(colors.magenta, styles.bold, "License: MIT");
  print_c(colors.magenta, styles.bold, "Build: 81424 (development)");
}

const colors = {
  reset: "\x1b[0m",
  red: "\x1b[31m",
  yellow: "\x1b[33m",
  blue: "\x1b[34m",
  green: "\x1b[32m",
  cyan: "\x1b[36m",
  magenta: "\x1b[35m",
  white: "\x1b[37m",
  black: "\x1b[30m",
  normal: "\x1b[39m",
};

const styles = {
  bold: "\x1b[1m",
  underline: "\x1b[4m",
  reverse: "\x1b[7m",
  normal: "\x1b[22m",
};

function n_line() {
  console.log("");
}

function print_c(style, color, msg) {
  console.log(`${style}${color}${msg}${colors.reset}`);
}

aboutV12();

File: src/utils/about_v12.rs
use super::logutil::{
  colors,
  styles,
  n_line,
  print_c
};

pub(crate) fn about_v12() {
  let banner = r#"
____    ____  __   ___      .______       __    __  .__   __. .___________. __  .___  ___.  _______
\   \  /   / /_ | |__ \     |   _  \     |  |  |  | |  \ |  | |           ||  | |   \/   | |   ____|
 \   \/   /   | |    ) |    |  |_)  |    |  |  |  | |   \|  | `---|  |----`|  | |  \  /  | |  |__
  \      /    | |   / /     |      /     |  |  |  | |  . `  |     |  |     |  | |  |\/|  | |   __|
   \    /     | |  / /_     |  |\  \----.|  `--'  | |  |\   |     |  |     |  | |  |  |  | |  |____
    \__/      |_| |____|    | _| `._____| \______/  |__| \__|     |__|     |__| |__|  |__| |_______|
"#;
  print_c(colors::MAGENTA, styles::BOLD, banner);
  n_line();
  print_c(colors::MAGENTA, styles::BOLD, " About V12 - A simple, lightweight, and easy-to-use JavaScript Runtime written in Rust.\n");
  print_c(colors::MAGENTA, styles::BOLD, " Version: 0.5.2\n");
  print_c(colors::MAGENTA, styles::BOLD, " Author(s): Thoq\n");
  print_c(colors::MAGENTA, styles::BOLD, " License: MIT\n");
  print_c(colors::MAGENTA, styles::BOLD, " Build: 10424 (development)\n");
  n_line();
  print_c(colors::MAGENTA, styles::BOLD, " HELP:\n");
  print_c(colors::MAGENTA, styles::BOLD, "   Usage: V12 <script_path>.ts/.js, version [debug]\n");
}
